src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getFruit() == null) {
+            throw new RuntimeException("There are no "
+                    + fruitTransaction.getFruit() + " int the storage");
+        }
+        int amountInStore = FruitStorage.getFruits().get(fruitTransaction.getFruit());

```

Comment:
There is a potential `NullPointerException` here. If `FruitStorage.getFruits().get(fruitTransaction.getFruit())` returns null because the fruit is not in the storage, this will cause a `NullPointerException` when unboxing. You should check if the fruit is present in the storage before attempting to get its amount.